java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -6080,6 +6075,68 @@ void testAllFilteredFromValidity() {
     }
   }
 
+  @Test
+  void testRemoveNullMasksIfNeeded() {
+    ListType nestedType = new ListType(true, new StructType(false,
+        new BasicType(true, DType.INT32),
+        new BasicType(true, DType.INT64)));
+
+    List data1 = Arrays.asList(10, 20L);
+    List data2 = Arrays.asList(50, 60L);
+    HostColumnVector.StructData structData1 = new HostColumnVector.StructData(data1);
+    HostColumnVector.StructData structData2 = new HostColumnVector.StructData(data2);
+
+    //First we create ColumnVectors
+    try (ColumnVector nonNullVector0 = ColumnVector.fromBoxedInts(1, 2, 3);
+         ColumnVector nonNullVector2 = ColumnVector.fromStrings("1", "2", "3");
+         ColumnVector nonNullVector1 = ColumnVector.fromLists(nestedType,
+             Arrays.asList(structData1, structData2),
+             Arrays.asList(structData1, structData2),
+             Arrays.asList(structData1, structData2))) {
+      //Then we take the created ColumnVectors and add validity masks even though the nullCount = 0
+      try (ColumnVector cv0 = new ColumnVector(nonNullVector0.type, nonNullVector0.rows,
```

Comment:
I personally would prefer to see a stand along method to add in the validity to a ColumnView.

```
ColumnView replaceValidity(ColumnView original, DeviceMemoryBuffer validity) {
    assert(validity.length >= BitVectorHelper.getValidityAllocationSizeInBytes(original.rows));
    // get children
    try {
        return new ColumnView(original.type,
            original.rows,
            Optional<Long>.empty(),
            validity,
            original.getOffsets(),
            children);
    } finally {
          // close children
    }   
}
```

Then we can insert this into children columns if we want.  We can copy it back out to a ColumnVector if we need to.  It is just a lot more flexible and cleaner.

Reply:
Do you have concerns with making this a instance method on ColumnView?

Reply:
nm, I misunderstood